File: app/src/main/java/me/askfix/api/askfix/centrifuge/AppPublishHandler.java
package me.askfix.api.askfix.centrifuge;

import centrifuge.PublishEvent;
import centrifuge.PublishHandler;
import centrifuge.Subscription;

public class AppPublishHandler implements PublishHandler {

    private final OnPublishListener listener;

    public AppPublishHandler(OnPublishListener listener) {
        this.listener = listener;
    }

    @Override
    public void onPublish(final Subscription sub, final PublishEvent event) {
        listener.onPublish(sub, event);
    }
}

File: app/src/main/java/me/askfix/api/askfix/centrifuge/AppPublishHandler.kt
package me.askfix.api.askfix.centrifuge

import android.app.Activity
import android.content.Context
import android.view.View
import android.widget.TextView

import centrifuge.PublishEvent
import centrifuge.PublishHandler
import centrifuge.Subscription
import me.askfix.api.askfix.MainActivity
import me.askfix.api.askfix.R

class AppPublishHandler(context: Context) : PublishHandler {
    private var context: MainActivity = context as MainActivity

    override fun onPublish(sub: Subscription, event: PublishEvent) {
        context.runOnUiThread {
            val tv = (context as Activity).findViewById<View>(R.id.tvConnectStatus) as TextView
            val data = String(event.data)
            tv.text = "${context.getString(R.string.new_publication)}${sub.channel()}: $data"
        }
    }
}